artifactory.py

```diff
@@ -1579,6 +1579,59 @@ def find_permission_target(self, name):
             return obj
         return None
 
+    def get_users(self):
+        request_url = self.drive + "/api/security/users"
+        r = self.session.get(request_url, auth=self.auth)
+        r.raise_for_status()
+        response = r.json()
+        users = []
+        for i in response:
+            user = User(self, i["name"])
+            user.read()
+            users.append(user)
+        return users
+
+    def get_groups(self):
+        request_url = self.drive + "/api/security/groups"
+        r = self.session.get(request_url, auth=self.auth)
+        r.raise_for_status()
+        response = r.json()
+        groups = []
+        for i in response:
+            group = Group(self, i["name"])
+            group.read()
+            groups.append(group)
+        return groups
+
+    def get_repositories(self):
+        request_url = self.drive + "/api/repositories"
+        r = self.session.get(request_url, auth=self.auth)
+        r.raise_for_status()
+        response = r.json()
+        repositories = []
+        for i in response:
+            if i["type"] == "LOCAL":
```

Comment:
All functions look very similar except `get_repositories`. May be we could collapse to a one function and rewrite code a little bit to don't repeat?
```python
def get_groups(self, lazy: bool):
    return self._get_all(url="api/security/groups", key="name", cls=Group, lazy=lazy)
```


For `get_repositories ` we can write simple factory, may be
```python
# dohq_artifactory/admin.py
@staticmethod
def Repository.create(self, data):
    if ....

# here.py
def get_repositories(self):
    return self._get_all(url="api/security/repositories", func=Repository.create, lazy=lazy)
```

and impliment `get_all` like this
```python 
def _get_all(...):
   ... 
   if cls: # pass just key
      ... cls(data[key])
  if func: # pass full dict
         func(data)
```